src/frontend/Ast_to_Mir.ml

```diff
@@ -198,16 +210,16 @@ let constrainaction_fname c =
     | Unconstrain -> FnUnconstrain )
 
 type decl_context =
-  { dread: readaction option
+  { dread: ioaction option
   ; dconstrain: constrainaction option
   ; dadlevel: autodifftype }
 
 let rec unsizedtype_to_string = function
   | UMatrix -> "matrix"
   | UVector -> "vector"
   | URowVector -> "row_vector"
-  | UReal -> "real"
-  | UInt -> "int"
+  | UReal -> "scalar"
```

Comment:
Is this what Stan currently says? I always find "scalar" such an ambiguous word. In my head it implies a contrast with a vector, but then you still don't quite know that it's continuous rather than discrete

Reply:
yep